common-web-app-guide.md

```diff
@@ -0,0 +1,146 @@
+# ODL Docker/Django/Webpack Web App Guide
+
+**SECTIONS**
+1. [Initial Setup](#initial-setup)
+2. [Running and Accessing the App](#running-and-accessing-the-app)
+3. [Troubleshooting](#troubleshooting)
+
+
+# Initial Setup
+
+### Major Dependencies
+- Docker
+  - _[OSX only]_ Recommended OSX install method: [Download from Docker website](https://docs.docker.com/mac/)
+- docker-compose
+  - Recommended install: pip (`pip install docker-compose`)
+- _[OSX only]_
+  - Virtualbox (https://www.virtualbox.org/wiki/Downloads)
+  - Node/NPM 
+    - Recommended install method: [Installer on Node website](https://nodejs.org/en/download/)
+    - *NOTE: No specific versions have been chosen yet for NPM.*
+  - Yarn
+    - Some of our projects ship with a helpful yarn installation script: `sudo node ./scripts/install_yarn.js`
+    - If the the install script doesn't exist, install by running: `npm install -g yarn@0.27.0` (or whatever version is specified for yarn in `package.json`; use `sudo` as necessary)
+
+All commands in this guide should be run from the root directory of your project's repository (unless specified otherwise).
+
+### _[OSX only]_ Create a Docker VM for this app
+
+Replace `MACHINENAME` with a name that indicates the project (e.g.: `mm` or `micromasters` for the MicroMasters project). Allowed characters for the machine name: alphanumeric characters, periods, and dashes.
+
+    docker-machine create --driver virtualbox MACHINENAME
+    docker-machine start MACHINENAME
+    # 'docker-machine env MACHINENAME' prints export commands for important environment variables
+    eval "$(docker-machine env MACHINENAME)"
+
+### Docker Container Setup
+
+#### 1) Create your ``.env`` file
+ 
+    cp .env.example .env
+
+#### 2) Build the containers
+Run this command:
+
+    docker-compose build
+
+*NOTE: You will also need to run this command whenever ``requirements.txt`` or ``test_requirements.txt`` change.*
+
+#### 3) Create data structures
+Create the database tables from the Django models:
+
+    docker-compose run web ./manage.py migrate
+
+#### 3) Create a superuser
+You will be prompted for the username and some other details for this user.
+
+    docker-compose run web ./manage.py createsuperuser
+
+
+# Running and Accessing the App
+
+#### 1) _[OSX only]_ Make sure the right environment variables are set before running the container or executing any commands: 
+
+```
+eval "$(docker-machine env MACHINENAME)"
+```
+
+#### 2) _[OSX only]_ Set up and run the webpack dev server on your host machine
+
+In a separate terminal tab, use the webpack helper script to install npm modules and run the dev server:
+
+    ./webpack_dev_server.sh --install
+
+*NOTE: The ``--install`` flag is only needed if you're starting up for the first time, or if updates have been made to the packages in ``./package.json``.*
+
+#### 3) Run the container
+Start all the services that are required to run the app:
+
+    docker-compose up
+
```

Comment:
I thought that the docker `watch` container was supposed to detect you were on a mac and exit, so that it wouldn't interfere with running it natively. I just looked over the code for open-discussions, anyway, and it looks like it checks an `$IS_OSX` environment variable - maybe that has to be manually set?

@noisecapella